NuRadioReco/detector/RNO_G/rnog_detector.py

```diff
@@ -1179,28 +1202,29 @@ def get_time_delay(self, station_id, channel_id, cable_only=False, use_stored=Tr
         time_delay = 0
         for key, value in signal_chain_dict["response_chain"].items():
 
-            if re.search("cable", key) is None and cable_only:
+            if re.search("cable", key) is None and re.search("fiber", key) and cable_only:
                 continue
 
+            weight = value.get("weight", 1)
             if use_stored:
-                if "time_delay" not in value or "cable_delay" not in value:
+                if "time_delay" not in value and "cable_delay" not in value:
                     self.logger.warning(
                         f"The signal chain component \"{key}\" of station.channel "
-                        f"{station_id}.{channel_id} has no cable/time delay stored... Skip it")
+                        f"{station_id}.{channel_id} has no cable/time delay stored... (hence return time delay without it)")
                     continue
 
                 try:
-                    time_delay += value["time_delay"]
+                    time_delay += weight * value["time_delay"]
                 except KeyError:
-                    time_delay += value["cable_delay"]
+                    time_delay += weight * value["cable_delay"]
 
             else:
                 ydata = [value["mag"], value["phase"]]
                 response = Response(value["frequencies"], ydata, value["y-axis_units"],
                                     name=key, station_id=station_id, channel_id=channel_id,
                                     log_level=self.__log_level)
 
-                time_delay += response._get_time_delay()
+                time_delay += weight * response._calculate_time_delay()
```

Comment:
Nit-picky comment: It might be slightly nicer to pass `weight` to `response` directly and let the object take care of it in the calculation of the time delay. But I don't think there is a way in which the current manual application can result in an incorrect answer, so either way is fine.

Reply:
We can discuss that yes. But lets merge this first.